src/FieldArray.tsx

```diff
@@ -85,6 +85,23 @@ export const replace = (array: any[], index: number, value: any) => {
   copy[index] = value;
   return copy;
 };
+
+const copyArray = (array?: any[]) => {
+  if (!array) {
+    return [];
+  }
+  if (isFunction(array.slice)) {
+    return [...array];
+  }
+
+  // It's an array-like object, e.g. {2: "hello"}, let's create a sparse array from it.
+  const copy = new Array();
+  Object.keys(array).forEach(k => {
```

Comment:
Although this is fine, you can use [`Array.From(arrayLike, mapFn)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from)